jetstream/outcomes.py

```diff
@@ -0,0 +1,26 @@
+from typing import Dict
+
+import attr
+
+from . import external_config
+
+
+@attr.s(auto_attribs=True)
+class _OutcomesResolver:
+    """Fetches and resolves external outcome snippet definitions."""
+
+    @property
+    def data(self) -> Dict[str, external_config.ExternalOutcome]:
+
+        if data := getattr(self, "_data", None):
+            return data
+
+        external_configs = external_config.ExternalConfigCollection.from_github_repo()
```

Comment:
One last thing about this: this is a fairly expensive operation; can we do this once instead of cloning it separately here? Can we attach the resolver to an instantiated ExternalConfigCollection? I guess that implies we'd have to pass it into AnalysisSpec.resolve().

Reply:
You are right, it makes sense to reuse the ExternalConfigCollection collection here. Passing it into AnalysisSpec.resolve() seemed like the best thing to do.